### ISSUE TITLE
Notebook_5_gan tweaks, changes...

### ISSUE THREAD

**Issue by bryan-pardo** [interactiveaudiolab/course-deep-learning#5]
![image](https://user-images.githubusercontent.com/26678616/155122795-124f33e3-2138-44fa-9b31-9d32fafdc0fc.png)

**Comment by bryan-pardo** [comment]
This could use a bit of text labeling. I'd add it, but don't have access to the underlying file. What I imagine is...

z: Random noise vector input to G.
G(z): The output of the generator
x: an example of real data
picture of all the little numbers tiled in a square: The set of real examples
D(G(z)):  Not sure what to put. You have these vertical pink and blue arrows.....I have no idea what they are about.

**Comment by bryan-pardo** [comment]
<img width="1228" alt="Screen Shot 2022-02-22 at 5 49 49 AM" src="https://user-images.githubusercontent.com/26678616/155146363-c86b4fec-818f-4f81-9b23-66792da9972a.png">

**Comment by bryan-pardo** [comment]
Accidentally closed the issue. This next bit is about the DCGan example:

I didn't understand this docstring:
`        Unlike our MLP generator, we'll interpret our latent inputs as having three 
        (non-batch) dimensions: channels, height, and width. Our latents will start
        with only a nontrivial channel dimension, and we will gradually reshape them 
        to "spread out" over the height and width dimensions using transposed 
        convolutions. `
What is meant by 'will start with only a nontrivial channel dimension'? I don't get what is intended.

In the initialization of the DCGan, I don't know why the weight means and standard deviations were chosen. Is the reason for these in the paper? Could some explanation be added for those choices? I'm talking about this code...
` # initialize weights
        for m in self.modules():
            name = m.__class__.__name__

            if name.find('Conv') != -1:
                nn.init.normal_(m.weight.data, 0.0, 0.02)

            elif name.find('BatchNorm') != -1:
                nn.init.normal_(m.weight.data, 1.0, 0.02)
                nn.init.constant_(m.bias.data, 0)`

**Comment by bryan-pardo** [comment]
I wonder if the DC gan stuff could become the basis of a homework? Then maybe it isn't in the tutorial notebook....

**Comment by oreillyp** [comment]
Ok, I've pushed an update addressing everything through the FFMPEG issue (which I resolved by switching from .gif to .mp4). Regarding the DCGAN portion of the notebook, I agree that this could be homework rather than tutorial material -- the notebook is already fairly long. This would allow for things like weight initialization (as you mentioned above) and mode collapse to be the basis for homework questions. To this end I've temporarily removed the DCGAN sections of the notebook, under the assumption I'll be shaping it into homework questions at some point soon.
